File: 1]. DSA + CP/1]. DSA/1]. Data Structures/01]. Array/Python/_002)_Insert_Element.py
"""
Function to insert a new element at a given position

Takes the position of an array and an element as arguments
"""

arr = [1, 2, 3, 5]

def insert(arr, index, successor):
    for index in range(index, len(arr)):
        arr[index], successor = successor, arr[index]

    arr.append(successor)

print("Array before insertion:", arr)
# OUTPUT: [1, 2, 3, 5]
insert(arr, 3, 4) 
print("Array after insertion:", arr)
# OUTPUT: [1, 2, 3, 4, 5]

File: 1]. DSA + CP/1]. DSA/1]. Data Structures/01]. Array/Dart/_002)_Insert_Element.dart
void insert(arr, positon, element) {
  if (positon < 0 || positon > arr.length) {
    throw new Exception("Invalid position");
  }
  for (int i = positon; i < arr.length; i++) {
    var temp = arr[i];
    arr[i] = element;
    element = temp;
  }
  arr.add(element);
}

void main() {
  var arr = [1, 2, 3, 4, 5];
  print('Before Insertion: ${arr}');
  insert(arr, 2, 6);
  print('After Insertion: ${arr}');
}
